### ISSUE TITLE
Kafka keep retrying when broker is not available

### ISSUE THREAD

**Issue by snp7890** [quarkusio/quarkus#32438]
### Describe the bug

I am setting below properties in application. properties but it's not working when broker is working it is keep retrying n number times. I want to retry only trice. 

kafka. retry-attempts=3

### Expected behavior

It should retry 3 times

### Actual behavior

Keep retrying n number of times

### How to Reproduce?

Start zookeeper
Don't start kafka broker then start the application

**Comment by ozangunalp** [comment]
I think there is confusion between different retry options:

On the consumer side (for incoming channels)
`retry`, `retry-attempts` and `retry-max-wait` options configure the behavior to retry processing of consumed Kafka records. So if no record is consumed, because Kafka is unreachable, it doesn't have any effect.

On the producer side (for outgoing channels)
`retries` option configures the number of times send operation is retried if the error is recoverable. When the number of retries is exhausted, it'll be logged and the outgoing message will be nack'ed.

In addition to all this, Kafka channels create clients eagerly and if clients don't hold a connection to brokers, they'll keep retrying to establish a connection.

**Comment by snp7890** [comment]
I want to retry connection 3 times when kafka broker not available for incomming channel

**Comment by cescoffier** [comment]
This is handle by the internal Nero work client of Kafka. Last time I checked it was not configurable.

**Comment by ozangunalp** [comment]
Yes, the underlying Kafka client does the continuous connection retry, I don't think it is configurable.
The health checks with smallrye-health check the `connection-count` metrics attribute to verify that the client is connected.

What can be done is to close the client prematurely, if no connection-count is found. However, it'll permanently put the channel in an error state.

**Comment by snp7890** [comment]
How to achieve this by code if not configurable?

**Comment by cescoffier** [comment]
I don't believe you can influence that behavior. It's the network client of Kafka, not something you (even we) can change.
@ozangunalp's approach looks promising. Monitor the number of connections in the metrics and if it's still 0 after a given duration - fail explicitly the application. This can be done using a health check probe for instance.
